src/helpers/useFocusItemHelper.ts

```diff
@@ -0,0 +1,171 @@
+import _ from 'lodash';
+import {RefObject, useState, useCallback, useEffect, useRef} from 'react';
+import {LayoutChangeEvent, ScrollView, FlatList} from 'react-native';
+import {Constants} from '.';
+
+export enum OffsetType {
+  CENTER = 'CENTER',
+  DYNAMIC = 'DYNAMIC',
+  LEFT = 'LEFT',
+  RIGHT = 'RIGHT'
+}
+
+export type Props = {
+  /**
+   * A reference to the ScrollView (or FlatList) which the items are in
+   */
+  scrollViewRef: RefObject<ScrollView | FlatList>;
+  /**
+   * The number of items
+   */
+  itemsCount: number;
+  /**
+   * The selected item's index
+   */
+  selectedIndex?: number;
+  /**
+   * Where would the item be located (default to CENTER)
+   */
+  offsetType?: OffsetType;
+  /**
+   * Add a margin to the offset (default to true)
+   * This gives a better UX
+   * Not relevant to OffsetType.CENTER
+   */
+  addOffsetMargin?: boolean;
+  /**
+   * How much space (padding \ margin) is there on he left\right of the items
+   */
+  sideSpacing?: number;
```

Comment:
I liked innerSpacing, maybe this one can be outerSpacing? 
```suggestion
  outerSpacing?: number;
```

Reply:
Done